core/src/main/java/io/questdb/cairo/sql/Function.java

```diff
@@ -250,6 +242,10 @@ default boolean isUndefined() {
         return getType() == ColumnType.UNDEFINED;
     }
 
+    default void offerStateTo(Function that) {
```

Comment:
Why don't we simply include the source argument into `init(SymbolTableSource symbolTableSource, SqlExecutionContext executionContext)`?

Reply:
as discussed, on slack, this method required a little less code refactoring and makes changes more localised to where they are required

Reply:
One more downside of splitting single action (initialization) into two methods. Resetting the state becomes more complicated as flags are required.